Thorough notes for contributors

## Basic Issue
The `README` currently says:  
> 1.  Goto `desktop-app` folder
> 2. Run `yarn dev`. This will start the app for local development with live reloading.

This gave me errors that yarn wasn't installed, then that modules weren't found. 

## Desired Result
Step-by-step instructions that cover installing dependencies, including:
- installing npm
- installing yarn
- Any npm setup that needs done before `yarn dev`
- Anything else I'm missing

It would be nice to have a contributors file that details these things. 

Other guidance would be nice, too, like if we need to run tests or any build steps or something? The process for sending it back upstream wouldn't hurt (do we PR into the master branch? I'm guessing yes) For the most part, clarity is just nice to have imo, even if things seem obvious.

## Extended Notes
First error was that yarn was not found, so I tried to `sudo dnf install yarn` before realizing I needed to go through NPM, which thankfully I had installed when doing a tutorial a month ago. I'm a PHP developer, primarily. 

From the `desktop-app` folder, I ran `yarn dev` and got errors like:
- Error: Cannot find module '@babel/register'
- Error: Cannot find module '@babel/core'
- Error: Cannot find module '@babel/plugin-transform-classes'
- Cannot find module 'react-hot-loader/babel'

I ended up running these commands:
```bash
npm install --save-dev @babel/register
npm install --save-dev @babel/core
npm install --save-dev @babel/plugin-transform-classes
```

Then got the `react-hot-loader` error, and thought maybe I should just `npm install`, so I did a `git reset --hard mycommithash`, then just ran `npm install`. Still getting the `react-hot-load/babel` error. I tried deleting the `yarn.lock` file & the `node_modules` folder too (but not sure if I'm supposed to)

I'm guessing `npm install --save-dev` is what I needed? But then I got errors from that too (without resetting everything)

I was trying to work on https://github.com/responsively-org/responsively-app/issues/433, which appears to be quite a simple fix, but now I'm not sure what to do to get the dev environment working

--- a/README.md
+++ b/README.md
@@ -77,8 +77,10 @@ If you face any problems while using the application, please open an issue here 
 Here is the roadmap of the desktop app - https://github.com/responsively-org/responsively-app/projects/12?fullscreen=true.
 
 ## Contribute
-1. Goto `desktop-app` folder
-2. Run `yarn dev`. This will start the app for local development with live reloading.
+1. Ensure you have installed `node` and `yarn`
+2. Go to `desktop-app` folder
+3. Run `yarn` to install dependencies
+4. Run `yarn dev`. This will start the app for local development with live reloading.
 
 ## Get in touch
 

